### ISSUE TITLE
RoPE中对q和k的维度拆分不应该是按奇偶分吗？

### ISSUE THREAD

**Issue by Phoebe-ovo** [THUDM/ChatGLM-6B#463]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

在modeling_chatglm.py中
q1, q2 = query_layer.chunk(2, dim=(query_layer.ndim - 1))
k1, k2 = key_layer.chunk(2, dim=(key_layer.ndim - 1))

### Steps To Reproduce

None

### Environment

```markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
```
